src/shared/ui/Stepper/index.tsx

```diff
@@ -80,14 +86,15 @@ export function Stepper({
       shouldAutoSubmit: boolean;
     }) => {
       const step = stepRef.current;
-      const stepShift = (await onBeforeNextRef.current?.(step)) ?? 1;
+      const { stepShift = 1, shouldIgnoreUserActionTrack = false } =
+        (await onBeforeNextRef.current?.(step)) ?? {};
       const nextStep = step + stepShift;
 
       const moved = viewSliderRef.current?.next(stepShift);
 
       if (moved) {
         stepRef.current = nextStep;
-        onNextRef.current?.(nextStep, isForced);
+        onNextRef.current?.(nextStep, isForced, shouldIgnoreUserActionTrack);
```

Comment:
@anq83 I am sorry but I don't think it was a good idea to let the shared component know about "actions". The only things that Stepper should know and operate are steps and isForced.

@vmkhitaryanscn 
Please make it possible to pass a TPayload object to the function:

```
...
  onNext?: (
    step: number,
    isForced: boolean,
    payload: TPayload,
  ) => void;
  onBeforeNext?: (
    step: number,
  ) => Promise<{ stepShift: number; payload: TPayload }>;
...
const { stepShift = 1, payload } = (await onBeforeNextRef.current?.(step)) ?? {};
...
onNextRef.current?.(nextStep, isForced, payload);
```
 
Then pass the payload in the outer scope (in the ActivityStepper component)